controllers/ControllerDecorator.ts

```diff
@@ -1,13 +1,59 @@
 import 'reflect-metadata';
-import {Router} from 'express';
-import {RouteMethod, routesKey} from '../routes/RouteDecorators';
-import {HttpVerbNotSupportedError, DuplicateRouteDeclarationError} from '../errors/Errors';
+import {Router, Request, Response} from 'express';
+import {RouteMethod, ROUTES_KEY, RouteRegistration} from '../routes/RouteDecorators';
+import {
+    HttpVerbNotSupportedError,
+    DuplicateRouteDeclarationError,
+    ParameterConstructorArgumentsError,
+    WrongReturnTypeError,
+    RouteError,
+    RequiredParameterNotProvidedError,
+    ParameterParseError,
+    ParamValidationFailedError
+} from '../errors/Errors';
+import {Param, PARAMS_KEY, ParamType} from '../params/ParamDecorators';
+import {ErrorHandlerManager, ERRORHANDLER_KEY} from '../errors/ErrorHandlerDecorator';
+import httpStatus = require('http-status');
 
 let controllers: ControllerRegistration[] = [],
     definedRoutes = [];
 
+const primitiveTypes = [Object, String, Array, Number, Boolean],
+    nonJsonTypes = [String, Number, Boolean];
+
+const defaultErrorHandler = (req, res, err) => {
+    console.error(err);
+    res.status(httpStatus.INTERNAL_SERVER_ERROR).end();
+};
+
+let parseParam = (value: any, param: Param) => {
```

Comment:
I moved the default error handler, but I'd rather not move the parseParam function, since it's private only for this module